# jbrowse2_utils
Bash scripts for writing assemblies and tracks to JBrowse2

This should be self explanatory for the most part. 

Requirements
 * genometools
 * tabix
 * JBrowse2

As usual with JBrowse2
 * first create an assembly
 * then use the sort script to sort GFF and GTF annotations
 * Add those using 
 * Add further BAM tracks etc
